core/common/src/main/java/alluxio/resource/DynamicResourcePool.java

```diff
@@ -336,15 +323,15 @@ public T acquire(long time, TimeUnit unit)
    */
   @Override
   public void release(T resource) {
+    if (!mResources.containsKey(resource)) {
+      throw new IllegalArgumentException(
+          "Resource " + resource.toString() + " was not acquired from this resource pool.");
+    }
+    ResourceInternal<T> resourceInternal = mResources.get(resource);
```

Comment:
It won't be removed because the resource won't be removed unless it is added to mResourcesAvailable (i.e. done with releasing). A comment is added.